torch/quantization/_correct_bias.py

```diff
@@ -0,0 +1,292 @@
+    '''
+    float_module_children = {}
+    for name, mod in float_model.named_children():
+        float_module_children[name] = mod
+
+    reassign = {}
+    for name, mod in quantized_model.named_children():
+        if name not in float_module_children:
+            continue
+
+        float_mod = float_module_children[name]
+
+        if type(float_mod) not in white_list:
+            add_shadow(float_mod, mod, white_list)
+
+        if type(float_mod) in white_list:
+            reassign[name] = ShadowModule(float_mod, mod)
+
+    for key, value in reassign.items():
+        quantized_model._modules[key] = value
+
+def remove_hooks(model, white_list=_supported_modules):
+    ''' Removes all the hooks from the whitelisted submodules
+    '''
+    for name, module in model.named_modules():
+        if type(module) in white_list:
+            if hasattr(module, 'activation_post_process'):
+                for key in module._forward_hooks:
+                    del module._forward_hooks[key]
+                del module.activation_post_process
+            if hasattr(module, 'activation_pre_process'):
+                for key in module._forward_pre_hooks:
+                    del module._forward_pre_hooks[key]
+                del module.activation_pre_process
+
+def remove_shadow(model):
+    ''' Goes through the submodules of a quantized model and replaces every
+    shadowModule with just the quantized submodule it contains
+    '''
+    reassign = {}
+    # logging occurences of shadowModules
+    for name, module in model.named_modules():
+        if isinstance(module, ShadowModule):
+            reassign[name] = module.quantized_module
+    # removing occurences of shadowModules
+    for name, module in reassign.items():
+        name_split = name.split('.')
+        get_parent_module(model, name)._modules[name_split[-1]] = reassign[name]
+
+def sequential_bias_correction(float_model, quantized_model, img_data, white_list=_supported_modules):
+    ''' Applies bias correction on the whitelisted submodules of the quantized model
+    After adding shadow modules and MeanLoggers to the quantized model the img_data will get reshaped
+    into one large batch and then this batch will be ran through the quantized model
+
+    at every whitelisted submodule during this execution, bias correction will be applied
+    on the quantized submodule within the shadowModule, before moving to the next submodule
+    '''
+    # adding hooks
+    # setting up qconfigs to add post hook
+    qconfig_debug = torch.quantization.QConfig(activation=MeanLogger, weight=None)
+    float_model.qconfig = qconfig_debug
+    quantized_model.qconfig = qconfig_debug
+    # calling prepare with prehook param, and throwing the whitelist to make sure supported modules
+    # get the qconfig and thus post hook
+    torch.quantization.prepare(float_model, inplace=True, white_list=white_list, prehook=MeanLogger)
```

Comment:
You should just create float and quantized models as usual and then use the numeric suite API to insert shadow modules. You do not need prehook or MeanLogger here.